### PR TITLE
CrystalAura rotation changes bruh

### PR DESCRIPTION
**Pull request by supakeks**
Default rotation settings send 2 times less packets now

Added an option to disable the remove entity thing as that's what was desyncing the fuck out of players on ncp servers (still on by default)

Also made the rotation resets only be sent when disabling because sending them after breaking/placing is packet spamming

Rotation Lock setting can be useful for servers with way more stricter rotation checks that may require you to look at a specific part of the support block

### CODE REVIEW

File: src/main/java/minegame159/meteorclient/modules/combat/CrystalAura.java [hunk 1]
```diff
@@ -422,11 +458,19 @@ public void onActivate() {
     @Override
     public void onDeactivate() {
         assert mc.player != null;
-        if (preSlot != -1) mc.player.inventory.selectedSlot = preSlot;
+        if (switchBack.get()) if (preSlot != -1) mc.player.inventory.selectedSlot = preSlot;
         for (RenderBlock renderBlock : renderBlocks) {
             renderBlockPool.free(renderBlock);
         }
         renderBlocks.clear();
+        if (target != null && resetRotations.get()) {
```
Comment (squidoodly) [inline src/main/java/minegame159/meteorclient/modules/combat/CrystalAura.java:466]: What does this actually do? It seems to just set where the player is looking to where they are already looking.

File: src/main/java/minegame159/meteorclient/modules/combat/CrystalAura.java [hunk 2]
```diff
@@ -295,6 +305,32 @@
...
+    private final Setting<RotationMode> rotationMode = sgRotations.add(new EnumSetting.Builder<RotationMode>()
+            .name("rotation-mode")
+            .description("When to rotate when using Crystal Aura.")
+            .defaultValue(RotationMode.Place)
+            .build()
+    );
+
+    private final Setting<Double> rotationLock = sgRotations.add(new DoubleSetting.Builder()
+            .name("place-rotation-lock")
+            .description("Where to look at when placing crystals.")
```
Comment (squidoodly) [inline src/main/java/minegame159/meteorclient/modules/combat/CrystalAura.java:319]: Would be nice to get a better description of this.

File: src/main/java/minegame159/meteorclient/modules/combat/CrystalAura.java [hunk 3]
```diff
@@ -422,11 +458,19 @@ public void onActivate() {
     @Override
     public void onDeactivate() {
         assert mc.player != null;
-        if (preSlot != -1) mc.player.inventory.selectedSlot = preSlot;
+        if (switchBack.get()) if (preSlot != -1) mc.player.inventory.selectedSlot = preSlot;
```
Comment (squidoodly) [inline src/main/java/minegame159/meteorclient/modules/combat/CrystalAura.java:461]: Can this be made into one if statement?

**Review by squidoodly — CHANGES_REQUESTED** [review]

File: src/main/java/minegame159/meteorclient/modules/combat/CrystalAura.java (hunk 1 above)
Comment (supakeks) [reply src/main/java/minegame159/meteorclient/modules/combat/CrystalAura.java:466]: It's doing what the return mode used to do, reset your rotation server-side as you would still be looking at the support block/end crystal

**Pull request merged**